### PR TITLE
enhancement!(http_server source,http_client source): Rename http sources to provide better clarity

### LINKED ISSUE #14522: Rename `http` => `http_server` and `http_scrape` => `http_client`
# Context

Historically the `http` source has been a _source_ of confusion due to both it's ambiguous name as well as somewhat unclear description:

```markdown
Collect logs emitted by an `HTTP` server
```

The `http` source is, in fact, a server that runs as part of Vector which receives requests from external HTTP clients, converting the request body into events.

We've recently added a `http_scrape` source which makes requests to an external HTTP server, a generalized form of the `prometheus_scrape` source.

As the `http_scrape` source has yet to make its stable appearance, we have the opportunity to rename it as a non-breaking change. I propose renaming these two sources along the well known client/server divide, this will be a more obvious naming strategy for these components and ideally cause less user confusion.

# Implementation

- Rename `http` source to `http_server`
- Rename `http_scrape` source to `http_client`
- Rename `Util::Http` to `Util::HttpServer`
  - This code is shared by some, but not all, HTTP server based sources
- Rename `Util::HttpScrape` to `Util::HttpClient` 
  - This code is shared by the `prometheus_scrape` source

# Out-of-Scope

- Renaming the `prometheus_scrape` source
- Renaming the `prometheus_exporter` sink
- Renaming the `http` sink

The Prometheus ecosystem commonly uses both scrape (`scrape_interval`, `scrape_configs`, etc.) and exporter ([exporters](https://prometheus.io/docs/instrumenting/exporters/)), leading to a less confusing UX. The `http` sink (currently) has no analogous component, and we could consider renaming it when implemented or as part of another issue that enforces general naming standardization.

### PR DESCRIPTION
**Pull request by spencergilbert**
- chore: rename sources-http feature to sources-http_server
- chore: rename internal usage of http source to http_server
- feature(http_server source): add alias for old http source

Closes #14522

### CODE REVIEW

**Comment by jszwedko** [comment]
> NOTE: The `alias` behavior allows existing configs using `http` to continue to run, but `list` and `generate` won't print/recognize the old aliased name.

🤔 I'd expect `list` to not show it, but it is odd that `generate` doesn't recognize it.

**Comment by spencergilbert** [comment]
> 🤔 I'd expect `list` to not show it, but it is odd that `generate` doesn't recognize it.

It's likely due to the changes in how those are wired up with the new `configurable` trait. I personally don't have an issue with it, as I'd rather newly generated config to always produce non-deprecated names and options.

**Comment by jszwedko** [comment]
Yeah, it's not a big deal since `vector generate` shouldn't really be used in any sort of automation (hopefully 😨 ).

File: src/sources/http_client/mod.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#[cfg(feature = "sources-http_client")]
+pub mod scrape;
```
Comment (neuronull) [inline src/sources/http_client/mod.rs:2]: should this file (module) be `client` instead of `scrape` ?

File: src/sources/http_client/scrape.rs [hunk 2]
```diff
@@ -33,10 +33,10 @@ use vector_core::{
...
 };
 
-/// Configuration for the `http_scrape` source.
-#[configurable_component(source("http_scrape"))]
+/// Configuration for the `http_client` source.
+#[configurable_component(source("http_client"))]
 #[derive(Clone, Debug)]
-pub struct HttpScrapeConfig {
+pub struct HttpClientConfig {
     /// Endpoint to scrape events from. The full path must be specified.
```
Comment (neuronull) [inline src/sources/http_client/scrape.rs:40]: Wonder if we should rephrase this as well to something like "Endpoint to gather events from. <...>" ?
Same goes for the cue file.

Comment (spencergilbert) [reply src/sources/http_client/scrape.rs:40]: 👍 gather/collect is a good change there

File: src/sources/http_client/scrape.rs [hunk 3]
```diff
@@ -33,10 +33,10 @@ use vector_core::{
     event::Event,
 };
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sources/http_client/scrape.rs:35]: I swear I searched for `scrape` and everything was pretty much pointed to the utils mod I wanted to punt on for now, but I'll do a second pass.

**Comment by spencergilbert** [comment]
> Also, I think the cue files need to be updated ? Or is that being handled in a separate PR ?

I/CLion definitely missed all the cue and md files for the website :/

**Review by jszwedko — COMMENTED** [review]
Code looks good! I expected to see some cue and markdown changes here though?

Also just a reminder to add it to the 0.25.0 milestone so we pick it in there.

**Comment by spencergilbert** [comment]
> Code looks good! I expected to see some cue and markdown changes here though?
> 
> Also just a reminder to add it to the 0.25.0 milestone so we pick it in there.

I added the _issue_ to the milestone but not the PR 😬 and yeah, totally missed the docs thinking CLion had made the changes for me.

**Comment by spencergilbert** [comment]
Still need to rename the internal metrics and `HttpScrapeContext` - but they collide with existing code (and fix merge conflicts).

**Review by jszwedko — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Hmm, actually I'm not seeing the documentation on the branch deploy for the new sinks 🤔

**Review by StephenWakely — APPROVED** [review]

**Comment by spencergilbert** [comment]
The enterprise soak tests seem to be failing, but not the other http ones... so I'm not sure what's happening there - but should fix before merging...

**Comment by jszwedko** [comment]
> > Hmm, actually I'm not seeing the documentation on the branch deploy for the new sinks 🤔
> 
> No sinks, just sources?
> 
> https://deploy-preview-14739--vector-project.netlify.app/docs/reference/configuration/sources/http_client/
> 
> https://deploy-preview-14739--vector-project.netlify.app/docs/reference/configuration/sources/http_server/

Doh, yes, forgot this was sources 😅

**Comment by jszwedko** [comment]
Can we leave https://deploy-preview-14739--vector-project.netlify.app/docs/reference/configuration/sources/http too but just mark it as deprecated and point to https://deploy-preview-14739--vector-project.netlify.app/docs/reference/configuration/sources/http_server? Or add a redirect?

**Review by neuronull — APPROVED** [review]

**Review by tobz — APPROVED** [review]
Most of the changes here make sense to me. 👍🏻

**Review by jszwedko — APPROVED** [review]
